### ISSUE TITLE
Transpilation of case statements without condition

### ISSUE THREAD

**Issue by theodorton** [geppetto-apps/crow#8]
Crystal allows (like Ruby), a conditionless case statement:

``` rb
case
when cond_a
  # Do something
when cond_b
  # Do something
else
  # Do something
end
```

I can't recall using this myself, or seeing this in the wild so I'm not sure whether transpilation should be supported. One option will be to transform the whole case statement to an if statement, but I'm afraid the syntax isn't just syntactic sugar and that the transformed statement would lose something in the process.

The same goes for handling type conditions:

``` rb
case object
when Foo
  # Object is of type Foo
when Bar
  # Object is of type Bar
end
```

**Comment by jhass** [comment]
Crystal itself just rewrites it to a `if/elsif` chain on the AST level at some point.

**Comment by theodorton** [comment]
Thank you @jhass, and hi :)

I guess that's good news. But transpiling all `case` statements to `if/elseif` will make the outputted code less readable. Maybe do some inspection of the `case` statement before transpilation to see whether it should transpile to `switch` or an `if`.

edit: some typos

**Comment by theodorton** [comment]
Closed by #10
